src/dotnet/commands/dotnet-run/RunCommand.cs

```diff
@@ -38,6 +38,34 @@ public int Start()
                 .ExitCode;
         }
 
+        public RunCommand(string configuration,
+            string framework,
+            bool noBuild,
+            string project,
+            IReadOnlyCollection<string> args)
+        {
+            Configuration = configuration;
+            Framework = framework;
+            NoBuild = noBuild;
+            Project = project;
+            Args = args;
+        }
+
+        public RunCommand MakeNewWithReplaced(string configuration = null,
```

Comment:
Agree on I can just use the public setter. However, I think since I already had constructor refactored I can just go head remove the public setter. Enforce immutability is good thing since you don't want to change the command like this after it is initialized